homeassistant/components/sia/config_flow.py

```diff
@@ -0,0 +1,153 @@
+        vol.Optional(CONF_ENCRYPTION_KEY): str,
+        vol.Required(CONF_PING_INTERVAL, default=1): int,
+        vol.Required(CONF_ZONES, default=1): int,
+        vol.Required(CONF_IGNORE_TIMESTAMPS, default=False): bool,
+        vol.Optional(CONF_ADDITIONAL_ACCOUNTS, default=False): bool,
+    }
+)
+
+ACCOUNT_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ACCOUNT): str,
+        vol.Optional(CONF_ENCRYPTION_KEY): str,
+        vol.Required(CONF_PING_INTERVAL, default=1): int,
+        vol.Required(CONF_ZONES, default=1): int,
+        vol.Required(CONF_IGNORE_TIMESTAMPS, default=False): bool,
+        vol.Optional(CONF_ADDITIONAL_ACCOUNTS, default=False): bool,
+    }
+)
+
+
+def validate_input(data: dict):
+    """Validate the input by the user."""
+    SIAAccount.validate_account(data[CONF_ACCOUNT], data.get(CONF_ENCRYPTION_KEY))
+
+    try:
+        ping = int(data[CONF_PING_INTERVAL])
+        assert 1 <= ping <= 1440
+    except AssertionError as invalid_ping:
+        raise InvalidPing from invalid_ping
+    try:
+        zones = int(data[CONF_ZONES])
+        assert zones > 0
+    except AssertionError as invalid_zone:
+        raise InvalidZones from invalid_zone
+
+
+class SIAConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for sia."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+    data = None
+
+    async def async_step_add_account(self, user_input: dict = None):
+        """Handle the additional accounts steps."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=ACCOUNT_SCHEMA,
+                errors={},
+            )
+
+    async def async_step_user(self, user_input: dict = None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=HUB_SCHEMA, errors={}
+            )
+        errors = {}
+        try:
+            validate_input(user_input)
+        except InvalidKeyFormatError:
+            errors["base"] = "invalid_key_format"
+        except InvalidKeyLengthError:
+            errors["base"] = "invalid_key_length"
+        except InvalidAccountFormatError:
+            errors["base"] = "invalid_account_format"
+        except InvalidAccountLengthError:
+            errors["base"] = "invalid_account_length"
+        except InvalidPing:
+            errors["base"] = "invalid_ping"
+        except InvalidZones:
+            errors["base"] = "invalid_zones"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        if errors:
+            return self.async_show_form(
+                step_id="user", data_schema=HUB_SCHEMA, errors=errors
+            )
+        self.update_data(user_input)
+        await self.async_set_unique_id(str(self.data[CONF_PORT]))
```

Comment:
Is `CONF_PORT` stable?  If they move this to another Home Assistant install will it remain the same?

Reply:
It is stable in the sense that you can only have one of these listening on a certain port, otherwise the server can't start, so we don't want multiple instances running on the same port, but it is not unique in the sense that the alarm and the server (this component) just have to agree on a port to use, but this protocol doesn't get any unique features of the alarm systems it communicates with (no uuid's, mac addresses or anything else) and it also doesn't care where the message comes from, other than for the time it takes to process and send a response. so in between two events the alarm could get a new IP address and this protocol doesn't care.

Reply:
We should take it out as the unique id as it doesn't appear to meet the unique id requirements as it looks like one of the items that falls into the unacceptable sources:   https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#unacceptable-sources-for-a-unique-id

For the entities we can use the config entry id for the unique id base https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-of-last-resort

Reply:
I'm not sure, I think at least in the config flow it should be checked in this way, because otherwise you get one of the two you put on the same port not starting, for the entity ids I agree we can use the config entry id as id.

Reply:
We should prevent creating multiple config entries with the same port with something like this https://github.com/home-assistant/core/blob/dev/homeassistant/components/harmony/config_flow.py#L89

Reply:
used that pattern and added the entry_id in the unique_id for the entity.